### PR TITLE
java-openliberty: add stack template variables

### LINKED ISSUE #699: java-openliberty:  Add stack.yaml variables (go templates): parentpomgroup, parentpomid
Pretty self-explanatory, see the java-spring-boot2 stack for examples.

### PR DESCRIPTION
**Pull request by awisniew90**
### Checklist:

- [x ] Read the [Code of Conduct](https://github.com/appsody/website/blob/master/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md).

- [ x] Followed the [commit message guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).

- [ x] Stack adheres to [Appsody stack structure](https://github.com/appsody/website/blob/master/content/docs/stacks/stacks-overview.md#stack-structure).

### Modifying an existing stack:

- [ x] Updated the stack version in `stack.yaml`

Add parentpomgroup and parentpomid template variables
Add stack developer section to readme

### Related Issues:
Fixes #699

### CODE REVIEW

**Comment by awisniew90** [comment]
@scottkurz - Updated the template pom to use `{{.stack.parentpomgroup}}` inside the starter groupid

File: incubator/java-openliberty/README.md [hunk 1]
```diff
@@ -118,6 +118,34 @@ The metrics endpoint is secured with a userid and password enabled through the c
...
+
+### Stack template variables
+
+Custom template variables are defined in the `stack.yaml` file. The Java Open Liberty stack includes the following that are propagated when the stack is packaged.
+
+* **libertyversion** - The version of the Open Liberty runtime to be used for development and production images. Without any other modifications to the Dockerfiles in this project, this version must be a quarterly release (x.0.0.3, x.0.0.6, x.0.0.9, x.0.0.12)
+
+* **parentpomgroup** - The group id used for the parent pom definition
+* **parentpomid** - The artifact id used for the parent pom definition
+* **parentpomrange** - The version range used within the template project definitions. i.e. [0.2, 0.3). This allows micro version updates to be automatically picked up by users of this stack.
```
Comment (scottkurz) [inline incubator/java-openliberty/README.md]: definitions. i.e. [0.2, 0.3). 
  > 
definitions, e.g. [0.2, 0.3).

File: incubator/java-openliberty/README.md [hunk 2]
```diff
@@ -118,6 +118,34 @@ The metrics endpoint is secured with a userid and password enabled through the c
...
+
+Additionally, the Java Open Liberty stack includes a set of custom variables and enforcement rules that can be leveraged when modifying the stack.
+
+### Stack template variables
+
+Custom template variables are defined in the `stack.yaml` file. The Java Open Liberty stack includes the following that are propagated when the stack is packaged.
+
+* **libertyversion** - The version of the Open Liberty runtime to be used for development and production images. Without any other modifications to the Dockerfiles in this project, this version must be a quarterly release (x.0.0.3, x.0.0.6, x.0.0.9, x.0.0.12)
+
+* **parentpomgroup** - The group id used for the parent pom definition
```
Comment (scottkurz) [inline incubator/java-openliberty/README.md]: .  We also use this to form the "prefix" of the groupid used by the default template artifact.

**Comment by awisniew90** [comment]
Hi @Kamran64  - This is ready to merge.

**Pull request merged**